src/deployment.ts

```diff
@@ -135,6 +141,13 @@ export class Deployment extends workload.Workload implements IScalable {
    */
   public readonly strategy: DeploymentStrategy;
 
+  /**
+   * Number of desired replicasets history.
+   *
+   * @default 10
+   */
+
+  readonly revisionHistoryLimit?: number;
```

Comment:
As a field of the deployment this shouldn't be optional, as it has a default value.

```suggestion
   */

  readonly revisionHistoryLimit: number;
```